### PR TITLE
Updates Burr tracking client to safely encode json

### PR DESCRIPTION
**Pull request by elijahbenizzy**
There are often missing characters -- this enables the user to safely encode them. They'll get replaced, but generally that's OK (and certainly better than failing).

Hard to test, but it doesn't introduce a regression so I'm happy enough for now.

### CODE REVIEW

**Review by skrawcz — APPROVED** [review]
no test case?

File: burr/tracking/client.py [hunk 1]
```diff
@@ -57,6 +57,12 @@ def _allowed_project_name(project_name: str, on_windows: bool) -> bool:
     return bool(re.match(pattern, project_name))
 
 
+def safe_json_write(line: str, f: io.TextIO):
```
Comment (skrawcz) [inline burr/tracking/client.py:60]: where is this used?

**Comment by elijahbenizzy** [comment]
> no test case?

See note. Could probably add weird bytes and serialize but its too convoluted for a small edge case IMO.
